airbyte-cdk/python/cdk-migrations.md

```diff
@@ -1,15 +1,19 @@
 # CDK Migration Guide
 
-## Importing classes
-Starting from 1.0.0, CDK classes and functions should be imported directly from `airbyte_cdk` (example: `from airbyte_cdk import HttpStream`). Lower-level `__init__` files are not considered stable, and will be modified without introducing a major release.
+## Upgrading to 3.0.0
+Version 3.0.0 of the CDK updates the `HTTPStream` class by reusing the `HTTPClient` under the hood.
 
-Introducing breaking changes to a class or function exported from the top level `__init__.py` will require a major version bump and a migration note to help developer upgrade.
+- `backoff_time` and `should_retry` methods are removed from HttpStream
+- `HttpStreamAdapterHttpStatusErrorHandler` and `HttpStreamAdapterBackoffStrategy` adapters are marked as `deprecated`
+- `raise_on_http_errors`, `max_retries`, `max_time`, `retry_factor` are marked as `deprecated`
 
-Note that the following packages are not part of the top level init because they require extras dependencies, but are still considered stable:
-- `destination.vector_db_based`
-- `source.file_based`
+### Migration of `should_retry` method
+In case the connector uses custom logic for backoff based on the response from the server, a new method `get_error_handler` should be implemented.
+This method should return instance of [`ErrorHandler`](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/error_handler.py).
 
-The `test` package is not included in the top level init either. The `test` package is still evolving and isn't considered stable.
+### Migration of `backoff_time` method
+In case the connector uses custom logic for backoff time calculation, a new method `get_backoff_strategy` should be implemented.
+This method should return instance(s) of [`BackoffStrategy`](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/backoff_strategy.py).
 
```

Comment:
I would add another section saying that the exceptions that exceptions from the requests lib should not be raised anymore when calling `read_records` and therefore:
* catching exception should be updated
* error messages might change

If we can give an example for that one too, that would be great!

Reply:
updated with text and link to migration for zendesk support.